spec/presenters/submission_presenter_spec.rb

```diff
@@ -1,68 +1,98 @@
+      is_expected.not_to be_a(Submission)
     end
   end
 
   describe "#average_rate" do
     describe "when submission is unrated" do
-      let!(:submission) { FactoryGirl.build(:submission) }
+      subject { submission_presenter.average_rate }
 
-      it { expect(subject.average_rate).to eq(nil) }
+      before { allow(submission).to receive(:rated?).and_return(false) }
+
+      it { is_expected.to eq(nil) }
     end
 
     describe "when submission is rated and has no rates" do
-      before do
-        allow(Setting).to receive(:get).and_return(
-          FactoryGirl.create(:setting, required_rates_num: 0))
-      end
+      subject { submission_presenter.average_rate }
 
-      let!(:submission) do
-        FactoryGirl.build(:submission, :rated, required_rates_num: 0)
+      before do
+        allow(submission).to receive(:rated?).and_return(true)
+        allow(rates).to receive(:count).and_return(0)
       end
 
-      it { expect(subject.average_rate).to eq(0) }
+      it { is_expected.to eq(0) }
     end
 
     describe "when submission is rated and has some rates" do
-      before do
-        allow(Setting).to receive(:get).and_return(
-          FactoryGirl.create(:setting, required_rates_num: 3))
-      end
+      subject { submission_presenter.average_rate }
 
-      let!(:submission) do
-        FactoryGirl.build(:submission, :rated, { required_rates_num: 3, rate_value: 2 })
+      before do
+        allow(submission).to receive(:rated?).and_return(true)
+        allow(rates).to receive(:count).and_return(4)
+        allow(rates).to receive(:sum).with(:value).and_return(10)
       end
 
-      it { expect(subject.average_rate).to eq(2) }
+      it { is_expected.to eq(2.5) }
     end
   end
 
-  describe "delegates methods to submissions_repository" do
-    let!(:submission) { FactoryGirl.create(:submission) }
+  describe "delegates methods to submission_repository" do
+    describe "#next_to_rate" do
+      let(:result) { double }
+      subject { submission_presenter.next_to_rate }
 
-    context "#next_to_rate" do
-      it do
-        expect(submissions_repository).to receive(:next_to_rate).with(submission.created_at)
-        subject.next_to_rate
+      before do
+        date = double
+        allow(submission).to receive(:created_at).and_return(date)
+        allow(submission_repository).to receive(:next_to_rate).with(date).and_return(result)
       end
+
+      it { is_expected.to eq(result) }
     end
 
-    context "#previous_to_rate" do
-      it do
-        expect(submissions_repository).to receive(:previous_to_rate).with(submission.created_at)
-        subject.previous_to_rate
+    describe "#previous_to_rate" do
+      let(:result) { double }
+      subject { submission_presenter.previous_to_rate }
+
+      before do
+        date = double
+        allow(submission).to receive(:created_at).and_return(date)
+        allow(submission_repository).to receive(:previous_to_rate).with(date).and_return(result)
```

Comment:
But this example with mocking is really nice. Because `previous_to_rate` is public.